artview/components/nav.py

```diff
@@ -0,0 +1,271 @@
+"""
+nav.py
+
+Icons used in this script were created by
+oxygenicons (http://www.oxygen-icons.org/)
+and distributed at the IconArchive (http://www.iconarchive.com) under
+the GNU Lesser General Public License.
+
+"""
+
+# Load the needed packages
+from functools import partial
+import os
+import numpy as np
+import pyart
+import time
+
+from ..core import Component, Variable, common, QtGui, QtCore, componentsList
+
+
+class FileNavigator(Component):
+    '''
+    Interface for executing :py:class:`pyart.filters.GateFilter`.
+    '''
+
+    Vradar = None  #: see :ref:`shared_variable`
+    Vgrid = None  #: see :ref:`shared_variable`
+    Vfilelist = None  #: see :ref:`shared_variable`
+
+    @classmethod
+    def guiStart(self, parent=None):
+        '''Graphical interface for starting this class.'''
+        kwargs, independent = \
+            common._SimplePluginStart("FileNavigator").startDisplay()
+        kwargs['parent'] = parent
+        return self(**kwargs), independent
+
+    def __init__(self, Vradar=None, Vgrid=None, Vfilelist=None,
+                 name="FileNavigator", parent=None):
+        '''Initialize the class to create the interface.
+
+        Parameters
+        ----------
+        name : string
+            FileNavigator instance window name.
+        parent : PyQt instance
+            Parent instance to associate to this class.
+            If None, then Qt owns, otherwise associated w/ parent PyQt instance
+        '''
+        super(FileNavigator, self).__init__(name=name, parent=parent)
+        self.central_widget = QtGui.QWidget()
+        self.setCentralWidget(self.central_widget)
+        self.layout = QtGui.QGridLayout(self.central_widget)
+
+        # Set up signal, so that DISPLAY can react to
+        # changes in radar or gatefilter shared variables
+        if Vradar is None:
+            self.Vradar = Variable(None)
+        else:
+            self.Vradar = Vradar
+        if Vgrid is None:
+            self.Vgrid = Variable(None)
+        else:
+            self.Vgrid = Vgrid
+        if Vfilelist is None:
+            self.Vfilelist = Variable(0)
+        else:
+            self.Vfilelist = Vfilelist
+
+        self.sharedVariables = {"Vradar": None,
+                                "Vgrid": None,
+                                "Vfilelist": None}
```

Comment:
navigator needs a NewFilelist. It need to respond to an update of filelist: update tool tips, check index, check that indexes don't overflow.